commands/indexDump.js

```diff
@@ -0,0 +1,130 @@
+/*
+ * Kuzzle, a backend software, self-hostable and ready to use
+ * to power modern apps
+ *
+ * Copyright 2015-2018 Kuzzle
+ * mailto: support AT kuzzle.io
+ * website: http://kuzzle.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* eslint-disable no-console */
+
+const
+  fs = require('fs'),
+  ColorOutput = require('./colorOutput'),
+  path = require('path'),
+  ndjson = require('ndjson'),
+  getSdk = require('./getSdk');
+
+//@TODO use 'recursive: true' with node.js 10
+function mkdirp (fullPath) {
+  const parts = fullPath.split(path.sep);
+
+  if (parts.length > 1) {
+    mkdirp(parts.slice(0, parts.length - 1).join('/'));
+  }
+
+  if (! fs.existsSync(fullPath)) {
+    fs.mkdirSync(fullPath);
+  }
+}
+
+function writeData(stream, data) {
+  return new Promise(resolve => {
+    if (! stream.write(data)) {
+      stream.once('drain', resolve);
+    }
+    else {
+      resolve();
+    }
+  });
+}
+
+async function dumpCollection (sdk, index, collection, batchSize, directoryPath) {
```

Comment:
Why are we using `async` here? We use promise chains in all the previous files...

Reply:
Writing this function without async/await is really a mess, this is the reason why I added babel to the CLI